decoder/reference_targets.go

```diff
@@ -740,14 +740,15 @@ func (d *PathDecoder) collectInferredReferenceTargetsForBody(addr lang.Address,
 
 		blockRef := reference.Target{
 			Addr:        blockAddr,
-			LocalAddr:   make(lang.Address, len(selfRefAddr)),
+			LocalAddr:   make(lang.Address, 0),
 			ScopeId:     bAddrSchema.ScopeId,
 			Type:        cty.Object(bodySchemaAsAttrTypes(bCollection.Schema.Body)),
 			Description: bCollection.Schema.Description,
 			DefRangePtr: blk.DefRange.Ptr(),
 			RangePtr:    blk.Range.Ptr(),
 		}
 		if bAddrSchema.DependentBodySelfRef && selfRefBodyRangePtr != nil {
+			blockRef.LocalAddr = make(lang.Address, len(selfRefAddr))
```

Comment:
I wonder if it would make more sense to keep creating the slice with a specific length and clear it instead, like:

```go
blockRef := reference.Target{
	Addr:        blockAddr,
	LocalAddr:   make(lang.Address, len(selfRefAddr)),
	...
}
if bAddrSchema.DependentBodySelfRef && selfRefBodyRangePtr != nil {
	copy(blockRef.LocalAddr, selfRefAddr)
	blockRef.LocalAddr = append(blockRef.LocalAddr, lang.AttrStep{Name: bType})
	blockRef.TargetableFromRangePtr = selfRefBodyRangePtr.Ptr()
} else {
	blockRef.LocalAddr = lang.Address{}
}
```

This would keep the "hot" path as is and only modify the slice, when dealing with JSON files.

Reply:
I'm not sure what's better.

The main motivation with precise length is reduce memory consumption, but I'm guessing that is thwarted anyway with the use of `append()` which will likely grow the allocation by more than what's needed for 1 element anyway? Not sure about that.

Also I am not sure what happens when a pre-allocated slice is set to an empty one, I'm assuming GC will take care of it, but the memory would have been allocated before subsequently being freed.

So unless we're looking to do performance optimisations - which I don't think was the goal of this PR - then I don't think it matters.

Reply:
👍 I'll keep the changes as they are as I'm not doing any performance optimizations.